recipes/pcl/all/conanfile.py

```diff
@@ -0,0 +1,202 @@
+        "with_openni2": [True, False],
+        "with_pcap": [True, False],
+        "with_png": [True, False],
+        "with_qhull": [True, False],
+        "with_rssdk": [True, False],
+        "with_tools": [True, False],
+        "with_usb": [True, False],
+        "with_vtk": [True, False],
+        "module_2d": [True, False],
+        "module_features": [True, False],
+        "module_filters": [True, False],
+        "module_geometry": [True, False],
+        "module_io": [True, False],
+        "module_kdtree": [True, False],
+        "module_keypoints": [True, False],
+        "module_ml": [True, False],
+        "module_octree": [True, False],
+        "module_registration": [True, False],
+        "module_sample_consensus": [True, False],
+        "module_search": [True, False],
+        "module_segmentation": [True, False],
+        "module_stereo": [True, False],
+        "module_surface": [True, False],
+        "module_tracking": [True, False],
+        "module_recognition": [True, False]
+    }
+    default_options = {
+        "shared": True,
+        "fPIC": True,
+        "boost:shared": True,
+        "with_cuda": False,
+        "with_davidsdk": False,
+        "with_dssdk": False,
+        "with_ensenso": False,
+        "with_opengl": True,
+        "with_openni": False,
+        "with_openni2": False,
+        "with_qhull": True,
+        "with_pcap": False,
+        "with_png": True,
+        "with_rssdk": False,
+        "with_tools": False,
+        "with_usb": False,
+        "with_vtk": False,
+        "module_2d": False,
+        "module_features": False,
+        "module_filters": False,
+        "module_geometry": True,
+        "module_io": True,
+        "module_kdtree": True,
+        "module_keypoints": False,
+        "module_ml": True,
+        "module_octree": True,
+        "module_recognition": False,
+        "module_registration": True,
+        "module_sample_consensus": False,
+        "module_search": False,
+        "module_segmentation": True,
+        "module_stereo": False,
+        "module_surface": True,
+        "module_tracking": False
+    }
+    requires = ("boost/[>1.69.0]",
+                "eigen/[>3.3.5]",
+                "flann/[>1.9.0]")
+    generators = "cmake"
+    exports = ["CMakeLists.txt"]
+
+    @property
+    def _major_minor_version(self):
+        major, minor, patch = self.version.split(".")
+        return ".".join([major, minor])
+
+    @property
+    def _source_subfolder(self):
+        return f"pcl-pcl-{self.version}"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        tools.replace_in_file("CMakeLists.txt", "add_subdirectory(pcl)", f"add_subdirectory({self._source_subfolder})")
+
+    def requirements(self):
+        if self.options.with_png:
+            self.requires("libpng/[>1.6.36]")
+        if self.options.with_qhull:
+            self.requires("qhull/[>7.3.0]")
+            self.options["qhull"].shared = False
```

Comment:
Why do you need qhull as static only?  I'm asking because this situation lies what the profile requests. You are forcing a different package ID from what is described in the profile. If it's a project limitation, I would prefer raising a ConanInvalidConfiguraion if qhull is shared.